string_encode.py

```diff
@@ -101,12 +101,19 @@ def run(self, edit, **kwargs):
 class Gzip64EncodeCommand(StringEncode):
 
     def encode(self, text):
-        return base64.b64encode(gzip.compress(bytes(text, 'utf-8'))).decode('ascii')
+        return str(base64.b64encode(gzip.compress(bytes(text, 'utf-8'))), 'ascii')
 
 class Gzip64DecodeCommand(StringEncode):
 
     def encode(self, text):
-        return gzip.decompress(base64.b64decode(text.rstrip('=') + '===')).decode('utf-8')
+        value = bytearray(text, 'ascii')
+        mod = len(value) % 4
+        if mod == 3:
+            value.extend(b"=")
+        elif mod == 2:
```

Comment:
No case needed for `mod == 1`?

Reply:
4 byte of base64 string represent 3 byte unencoded data. A single `=` indicates 2 bytes and `==` indicates 1 byte of unencoded data. Hence `mod == 1` most likely indicates invalid base64 string or already over-padded string.

Reply:
Ahh I see I was interpreting this the other way around, I was reviewing by phone and assumed this was padding the encode step (which also seemed unnecessary but I was putting that on pause).

I'm at the computer, and yes this works as intended when decoding data that *isn't* correctly padded. The only time the `mod == 1` comes up is when I try to decode *in-between* the byte.

```
dGVzdA==  => test
dGVzdA=   => test
dGVzdA    => test
dGVzd     => invalid
````